iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber_uds.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_HPP
+
+#include "iceoryx_hoofs/internal/posix_wrapper/mutex.hpp"
+#include "iceoryx_hoofs/internal/posix_wrapper/unix_domain_socket.hpp"
+
+#include "owl/types.hpp"
+
+#include <limits>
+#include <memory>
+#include <thread>
+
+
+namespace owl
+{
+namespace kom
+{
+/// @brief Class solely for benchmarking iceoryx against UNIX domain sockets
+template <typename T>
+class EventSubscriber<T, EventTransmission::UDS>
+{
+  public:
+    using SampleType = T;
+
+    static_assert(is_supported_topic<T>::value,
+                  "Topic must have specific members, look at TimestampTopic1Byte as an example!");
+
+    EventSubscriber(const ServiceIdentifier&, const InstanceIdentifier& instance, const EventIdentifier&) noexcept;
```

Comment:
Rule of 5

Reply:
Closing, as this file wil be moved to a separate repository.